### ISSUE TITLE
mathjs polluting Decimal prototype

### ISSUE THREAD

**Issue by ifeelfishy** [josdejong/mathjs#2222]
For some reason I need to use Decimal.js and mathjs at the same time in project. However mathjs pollutes the decimal prototype with its own toJSON function, this produces undesirable effects in serialization:

minimal code to reproduce
```
const Decimal = require('decimal.js')
const mathjs = require('mathjs')

const dec = new Decimal(0)
console.log(JSON.stringify(dec))
```
expected output
```
"0"
```

actual output
```
{"mathjs":"BigNumber","value":"0"}
```

Is there a way for creating the bignumber using decimal.js without polluting its prototype?

**Comment by josdejong** [comment]
Ai, that's not good. I wasn't aware of this issue, mathjs does do a `Decimal.clone(...)` and then adds some methods to the cloned version. I was assuming this would not impact the original `Decimal` instance but apparently it does.

Anyone able to look into possible solutions? Help would be welcome!

**Comment by ifeelfishy** [comment]
I can give it a shot, maybe extending the Decimal class and not cloning it (or extending the clone)...

**Comment by josdejong** [comment]
Thanks @ifeelfishy 👍

**Comment by m93a** [comment]
I tried messing around with the code that creates BigNumber and I must admit I have absolutely no idea what's going on there (see #2243).

**Comment by josdejong** [comment]
Thanks for having a look into this @m93a. You're indeed looking in the right file, [/src/type/bignumber/BigNumber.js](https://github.com/josdejong/mathjs/blob/develop/src/type/bignumber/BigNumber.js). 

In there, a clone is made of `Decimal`:

```js
const BigNumber = Decimal.clone({ precision: config.precision, modulo: EUCLID })
```

and then a couple of things is added to the prototype:

```js
BigNumber.prototype.type = 'BigNumber'
BigNumber.prototype.isBigNumber = true
BigNumber.prototype.toJSON = ...
BigNumber.prototype.fromJSON = ...
```
We do not want to make changes in the original `Decimal.prototype`, but apparently the cloned `BigNumber.prototype` is pointing to the original `Decimal.prototype`.

**Comment by ifeelfishy** [comment]
I had a look too and its very messy. The clone method in Decimal attaches the same prototype to the cloned constructor so that every instance created with the clones has the same prototype. One way to counter this behaviour may be to overwrite the constructor of BigNumber class to point to a copy of the decimal prototype. However when i tried this the tests failed, but the reason might be that some of the tests use Decimal.js directly and not BigNumber class...

**Comment by josdejong** [comment]
For reference: #2244

**Comment by josdejong** [comment]
Fixed now in `v9.4.3`. Thanks again @m93a
